eng/liveBuilds.targets

```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <CoreCLROSGroup Condition="'$(CoreCLROSGroup)' == ''">$(OSGroup)</CoreCLROSGroup>
-    <CoreCLRConfiguration Condition="'$(CoreCLRConfiguration)' == ''">$(Configuration)</CoreCLRConfiguration>
+    <CoreCLRConfiguration Condition="'$(CoreCLRConfiguration)' == ''">$(RuntimeConfiguration)</CoreCLRConfiguration>
```

Comment:
What if `RuntimeConfiguration` is not used? `CoreCLRConfiguration` would be empty in that case.

Reply:
Also the same should apply for MonoConfiguration, it should take RuntimeConfiguration as the configuration if it was passed globally.

Reply:
Hmm ... all that logic really needs to be hoisted up so both the live and CI builds share it. Let me do that.

Reply:
> What if RuntimeConfiguration is not used? CoreCLRConfiguration would be empty in that case.

When will this be the case though? All of our build scripts explicitly pass `/p:RuntimeConfiguration`.

Reply:
What if I just do: `build.cmd` (wanting to build everything on debug), or `build.cmd -c Release ` wanting to build all in Release. That will not pass down `RuntimeConfiguration`. That is actually why the runtime-live-build is failing in CI because, in those builds we don't use -runtimeConfiguration. 

Which I mentioned offline, we should exercise that in that build, we should have a leg for combining the subset's configurations around.

Reply:
If `RuntimeConfiguration` is empty, `Configuration` should be the one that is used.

Reply:
Ah okay, we're building up the argument array that way. 

In that case what is the default for the runtime configuration? When not specified `$(RuntimeConfiguration)` should be set to the default. To do otherwise introduces unnecessary complexity into our build scripts.

Reply:
Actually the right fix here, is to set `RuntimeConfiguration` to `Configuration` if `RuntimeConfiguration == ''` in our `Directory.Build.props` in the root and then always use `RuntimeConfiguration` for `CoreCLRConfiguration` and `MonoConfiguration`

Reply:
That was what I was thinking as well. Still don' tthink it's complete though because same problem applies: what if `$(Configuration)` is empty? 

Again this comes back to what are the defaults. We should fall back to them when the build doesn't specify them.

Reply:
`Configuration` will never be empty, `Configuration` is always passed down by arcade, default is `Debug`.

Reply:
https://github.com/dotnet/runtime/blob/master/eng/common/build.ps1#L3

Reply:
Also the property group in Subsets.props has the same issue fi RuntimeConfiguration is unspecified. Overall it's unclear why we repeat this property group between Directory.Build.props and SubSets.props. Seems much simpler to move this property group above the imports of Subsets.props and avoid repeating this logic

Reply:
This is the property group I'm referring to that should be moved https://github.com/dotnet/runtime/blob/master/Directory.Build.props#L55